### DISCUSSION TITLE
How to understand example about exceptions

### DISCUSSION THREAD

**Discussion by lszczygielek** [fluentassertions/fluentassertions#1731] [Q&A] [answered]
I'm looking at https://fluentassertions.com/exceptions/ and I understand example like:
```csharp
Action act = () => subject.Foo2("Hello");

act.Should().Throw<InvalidOperationException>()
    .WithInnerException<ArgumentException>()
    .WithMessage("whatever");
```

Then I look at paragraph about `async` and I see:
```csharp
Func<Task> act = () => asyncObject.ThrowAsync<ArgumentException>();
await act.Should().ThrowAsync<InvalidOperationException>();
await act.Should().NotThrowAsync();
```
And I don't know what it is. What is `asyncObject` and why `ThrowAsync` is called on it - it doesn't look like method, or I'm missing something?

In my tests I use:
```csharp
Func<Task<MyType>> func = () => myObject.MyMethodAsync(1).;
await func.Should().ThrowAsync<Exception>();
```
It works, but when I compare it with example from documentation, I have doubts if I'm doing OK.

**Comment by dennisdoomen** [comment, marked as answer]
In this example, the `asyncObject` is an instance of some type that has an `async` method names `ThrowAsync`.

**Reply by jnyrup** [reply]
I have written myself a note to see if this documentation can be even more clear.
